queries/markdown.scm

```diff
@@ -0,0 +1,19 @@
+(document) @textFragment
+
+(html_block) @comment
+
+;;!! * hello * stuff
+;;!  ^^^^^^^^^^^^^^^
+(list) @list
+
+;;!! # Title
+;;!    ^^^^^
+;;!! xxxxxxx
+;;!  -------
+(section
+    (atx_heading
+        (_)
+        heading_content: (_) @name
+    ) @_.removal
+    (#shrink-to-match! @name "^\\s*(?<keep>.*)$")
+) @_.domain
```

Comment:
Huh is it really this simple? I guess I only needed the leading sibling finder for when we were searching for a section heading of a specific level? If so I'd be tempted to just drop support for specific levels; I can't imagine anyone is using that 🤷‍♂️

Reply:
Oh I guess this is technically wrong. I believe "grand name" would be incorrect, because the domain should include all subheadings

Reply:
All the test passed. That was kind of my benchmark.

I would be happy to drop the specific section levels.

I think that is something we can worry about when we try to implement that logic?

Reply:
Already looks like subsections are nested
```scm
(document
  (section
    (atx_heading
      (atx_h1_marker)
      heading_content: (inline)
    )
    (paragraph
      (inline)
    )
    (section
      (atx_heading
        (atx_h2_marker)
        heading_content: (inline)
      )
      (paragraph
        (inline)
      )
    )
  )
)
```

Reply:
Oh amazing. I believe all that sibling logic is a holdover from the old markdown parser we used to use then. We should be able to drop most of that and do something much simpler now